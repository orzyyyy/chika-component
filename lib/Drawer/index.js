!function(e,t){if("object"==typeof exports&&"object"==typeof module)module.exports=t();else if("function"==typeof define&&define.amd)define([],t);else{var n=t();for(var r in n)("object"==typeof exports?exports:e)[r]=n[r]}}(window,function(){return function(e){var t={};function n(r){if(t[r])return t[r].exports;var o=t[r]={i:r,l:!1,exports:{}};return e[r].call(o.exports,o,o.exports,n),o.l=!0,o.exports}return n.m=e,n.c=t,n.d=function(e,t,r){n.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:r})},n.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},n.t=function(e,t){if(1&t&&(e=n(e)),8&t)return e;if(4&t&&"object"==typeof e&&e&&e.__esModule)return e;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,"default",{enumerable:!0,value:e}),2&t&&"string"!=typeof e)for(var o in e)n.d(r,o,function(t){return e[t]}.bind(null,o));return r},n.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return n.d(t,"a",t),t},n.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},n.p="",n(n.s=130)}({0:function(e,t,n){"use strict";e.exports=n(19)},10:function(e,t,n){"use strict";e.exports=function(e){var t=[];return t.toString=function(){return this.map(function(t){var n=function(e,t){var n=e[1]||"",r=e[3];if(!r)return n;if(t&&"function"==typeof btoa){var o=(u=r,"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,"+btoa(unescape(encodeURIComponent(JSON.stringify(u))))+" */"),i=r.sources.map(function(e){return"/*# sourceURL="+r.sourceRoot+e+" */"});return[n].concat(i).concat([o]).join("\n")}var u;return[n].join("\n")}(t,e);return t[2]?"@media "+t[2]+"{"+n+"}":n}).join("")},t.i=function(e,n){"string"==typeof e&&(e=[[null,e,""]]);for(var r={},o=0;o<this.length;o++){var i=this[o][0];null!=i&&(r[i]=!0)}for(o=0;o<e.length;o++){var u=e[o];null!=u[0]&&r[u[0]]||(n&&!u[2]?u[2]=n:n&&(u[2]="("+u[2]+") and ("+n+")"),t.push(u))}},t}},11:function(e,t,n){var r,o,i={},u=(r=function(){return window&&document&&document.all&&!window.atob},function(){return void 0===o&&(o=r.apply(this,arguments)),o}),a=function(e){var t={};return function(e,n){if("function"==typeof e)return e();if(void 0===t[e]){var r=function(e,t){return t?t.querySelector(e):document.querySelector(e)}.call(this,e,n);if(window.HTMLIFrameElement&&r instanceof window.HTMLIFrameElement)try{r=r.contentDocument.head}catch(e){r=null}t[e]=r}return t[e]}}(),c=null,f=0,l=[],s=n(22);function p(e,t){for(var n=0;n<e.length;n++){var r=e[n],o=i[r.id];if(o){o.refs++;for(var u=0;u<o.parts.length;u++)o.parts[u](r.parts[u]);for(;u<r.parts.length;u++)o.parts.push(m(r.parts[u],t))}else{var a=[];for(u=0;u<r.parts.length;u++)a.push(m(r.parts[u],t));i[r.id]={id:r.id,refs:1,parts:a}}}}function d(e,t){for(var n=[],r={},o=0;o<e.length;o++){var i=e[o],u=t.base?i[0]+t.base:i[0],a={css:i[1],media:i[2],sourceMap:i[3]};r[u]?r[u].parts.push(a):n.push(r[u]={id:u,parts:[a]})}return n}function y(e,t){var n=a(e.insertInto);if(!n)throw new Error("Couldn't find a style target. This probably means that the value for the 'insertInto' parameter is invalid.");var r=l[l.length-1];if("top"===e.insertAt)r?r.nextSibling?n.insertBefore(t,r.nextSibling):n.appendChild(t):n.insertBefore(t,n.firstChild),l.push(t);else if("bottom"===e.insertAt)n.appendChild(t);else{if("object"!=typeof e.insertAt||!e.insertAt.before)throw new Error("[Style Loader]\n\n Invalid value for parameter 'insertAt' ('options.insertAt') found.\n Must be 'top', 'bottom', or Object.\n (https://github.com/webpack-contrib/style-loader#insertat)\n");var o=a(e.insertAt.before,n);n.insertBefore(t,o)}}function h(e){if(null===e.parentNode)return!1;e.parentNode.removeChild(e);var t=l.indexOf(e);t>=0&&l.splice(t,1)}function v(e){var t=document.createElement("style");if(void 0===e.attrs.type&&(e.attrs.type="text/css"),void 0===e.attrs.nonce){var r=function(){0;return n.nc}();r&&(e.attrs.nonce=r)}return b(t,e.attrs),y(e,t),t}function b(e,t){Object.keys(t).forEach(function(n){e.setAttribute(n,t[n])})}function m(e,t){var n,r,o,i;if(t.transform&&e.css){if(!(i="function"==typeof t.transform?t.transform(e.css):t.transform.default(e.css)))return function(){};e.css=i}if(t.singleton){var u=f++;n=c||(c=v(t)),r=j.bind(null,n,u,!1),o=j.bind(null,n,u,!0)}else e.sourceMap&&"function"==typeof URL&&"function"==typeof URL.createObjectURL&&"function"==typeof URL.revokeObjectURL&&"function"==typeof Blob&&"function"==typeof btoa?(n=function(e){var t=document.createElement("link");return void 0===e.attrs.type&&(e.attrs.type="text/css"),e.attrs.rel="stylesheet",b(t,e.attrs),y(e,t),t}(t),r=function(e,t,n){var r=n.css,o=n.sourceMap,i=void 0===t.convertToAbsoluteUrls&&o;(t.convertToAbsoluteUrls||i)&&(r=s(r));o&&(r+="\n/*# sourceMappingURL=data:application/json;base64,"+btoa(unescape(encodeURIComponent(JSON.stringify(o))))+" */");var u=new Blob([r],{type:"text/css"}),a=e.href;e.href=URL.createObjectURL(u),a&&URL.revokeObjectURL(a)}.bind(null,n,t),o=function(){h(n),n.href&&URL.revokeObjectURL(n.href)}):(n=v(t),r=function(e,t){var n=t.css,r=t.media;r&&e.setAttribute("media",r);if(e.styleSheet)e.styleSheet.cssText=n;else{for(;e.firstChild;)e.removeChild(e.firstChild);e.appendChild(document.createTextNode(n))}}.bind(null,n),o=function(){h(n)});return r(e),function(t){if(t){if(t.css===e.css&&t.media===e.media&&t.sourceMap===e.sourceMap)return;r(e=t)}else o()}}e.exports=function(e,t){if("undefined"!=typeof DEBUG&&DEBUG&&"object"!=typeof document)throw new Error("The style-loader cannot be used in a non-browser environment");(t=t||{}).attrs="object"==typeof t.attrs?t.attrs:{},t.singleton||"boolean"==typeof t.singleton||(t.singleton=u()),t.insertInto||(t.insertInto="head"),t.insertAt||(t.insertAt="bottom");var n=d(e,t);return p(n,t),function(e){for(var r=[],o=0;o<n.length;o++){var u=n[o];(a=i[u.id]).refs--,r.push(a)}e&&p(d(e,t),t);for(o=0;o<r.length;o++){var a;if(0===(a=r[o]).refs){for(var c=0;c<a.parts.length;c++)a.parts[c]();delete i[a.id]}}}};var g,w=(g=[],function(e,t){return g[e]=t,g.filter(Boolean).join("\n")});function j(e,t,n,r){var o=n?"":r.css;if(e.styleSheet)e.styleSheet.cssText=w(t,o);else{var i=document.createTextNode(o),u=e.childNodes;u[t]&&e.removeChild(u[t]),u.length?e.insertBefore(i,u[t]):e.appendChild(i)}}},130:function(e,t,n){"use strict";n.r(t),n.d(t,"default",function(){return l});var r=n(0),o=n.n(r);n(252);function i(e){return(i="function"==typeof Symbol&&"symbol"==typeof Symbol.iterator?function(e){return typeof e}:function(e){return e&&"function"==typeof Symbol&&e.constructor===Symbol&&e!==Symbol.prototype?"symbol":typeof e})(e)}function u(){return(u=Object.assign||function(e){for(var t=1;t<arguments.length;t++){var n=arguments[t];for(var r in n)Object.prototype.hasOwnProperty.call(n,r)&&(e[r]=n[r])}return e}).apply(this,arguments)}function a(e,t){return!t||"object"!==i(t)&&"function"!=typeof t?function(e){if(void 0===e)throw new ReferenceError("this hasn't been initialised - super() hasn't been called");return e}(e):t}function c(e){return(c=Object.setPrototypeOf?Object.getPrototypeOf:function(e){return e.__proto__||Object.getPrototypeOf(e)})(e)}function f(e,t){return(f=Object.setPrototypeOf||function(e,t){return e.__proto__=t,e})(e,t)}var l=function(e){function t(){var e;return function(e,t){if(!(e instanceof t))throw new TypeError("Cannot call a class as a function")}(this,t),(e=a(this,c(t).apply(this,arguments))).handlePosition=function(t,n,r,o,i){var u=e.handleTransform(t),a=e.handleClientView(t,n,r,o,i).view;return"".concat(u,"(").concat(a,"px)")},e.handleTransform=function(e){return"left"===e||"right"===e?"translateX":"top"===e||"bottom"===e?"translateY":"transform error"},e.handleClientView=function(e,t,n,r,o){var i,u;return"left"===e||"right"===e?t=t||document.documentElement.clientWidth||document.body.clientWidth:"top"!==e&&"bottom"!==e||(t=t||document.documentElement.clientHeight||document.body.clientHeight),"left"===e||"top"===e?(i="operate"===o?n?r:0:n?0:-r,u=n?0:180):"right"!==e&&"bottom"!==e||(i="operate"===o?n?t-r-60:t-60:n?t-r:t,u=n?180:0),{view:i,iconDeg:u}},e.handleOperateClick=function(){var t=e.props,n=t.onChange,r=t.visible,o=r?180:0;n&&n(!r),e.setState({iconDeg:o})},e.render=function(){var t=e.props,n=t.children,r=t.width,i=void 0===r?.7*(document.body.clientWidth||document.documentElement.clientWidth):r,a=t.operaNode,c=t.direction,f=void 0===c?"left":c,l=t.visible,s=void 0!==l&&l,p=t.isOperateShow,d=void 0===p||p,y=t.clientView,h=void 0===y?0:y,v=t.style,b=e.handleClientView(f,h,s,i).iconDeg,m=o.a.createElement("div",{className:"operate-drawer",style:{transform:e.handlePosition(f,h,s,i,"operate"),background:"linear-gradient(to ".concat("left"===f?"right":"left",", rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0))")},onClick:e.handleOperateClick},o.a.createElement("div",{className:"scale-wrapper",style:{transform:"rotateZ(".concat(b,"deg)")}},a||o.a.createElement("img",{src:"./assets/arrow.png"}))),g=o.a.createElement("div",{className:"show-drawer",style:u({},{width:i},{transform:e.handlePosition(f,h,s,i)},v)},n);return o.a.createElement("div",{className:"Drawer"},g,d&&m)},e}return function(e,t){if("function"!=typeof t&&null!==t)throw new TypeError("Super expression must either be null or a function");e.prototype=Object.create(t&&t.prototype,{constructor:{value:e,writable:!0,configurable:!0}}),t&&f(e,t)}(t,r["Component"]),t}()},16:function(e,t,n){"use strict";
/*
object-assign
(c) Sindre Sorhus
@license MIT
*/var r=Object.getOwnPropertySymbols,o=Object.prototype.hasOwnProperty,i=Object.prototype.propertyIsEnumerable;e.exports=function(){try{if(!Object.assign)return!1;var e=new String("abc");if(e[5]="de","5"===Object.getOwnPropertyNames(e)[0])return!1;for(var t={},n=0;n<10;n++)t["_"+String.fromCharCode(n)]=n;if("0123456789"!==Object.getOwnPropertyNames(t).map(function(e){return t[e]}).join(""))return!1;var r={};return"abcdefghijklmnopqrst".split("").forEach(function(e){r[e]=e}),"abcdefghijklmnopqrst"===Object.keys(Object.assign({},r)).join("")}catch(e){return!1}}()?Object.assign:function(e,t){for(var n,u,a=function(e){if(null==e)throw new TypeError("Object.assign cannot be called with null or undefined");return Object(e)}(e),c=1;c<arguments.length;c++){for(var f in n=Object(arguments[c]))o.call(n,f)&&(a[f]=n[f]);if(r){u=r(n);for(var l=0;l<u.length;l++)i.call(n,u[l])&&(a[u[l]]=n[u[l]])}}return a}},19:function(e,t,n){"use strict";
/** @license React v16.8.6
 * react.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */var r=n(16),o="function"==typeof Symbol&&Symbol.for,i=o?Symbol.for("react.element"):60103,u=o?Symbol.for("react.portal"):60106,a=o?Symbol.for("react.fragment"):60107,c=o?Symbol.for("react.strict_mode"):60108,f=o?Symbol.for("react.profiler"):60114,l=o?Symbol.for("react.provider"):60109,s=o?Symbol.for("react.context"):60110,p=o?Symbol.for("react.concurrent_mode"):60111,d=o?Symbol.for("react.forward_ref"):60112,y=o?Symbol.for("react.suspense"):60113,h=o?Symbol.for("react.memo"):60115,v=o?Symbol.for("react.lazy"):60116,b="function"==typeof Symbol&&Symbol.iterator;function m(e){for(var t=arguments.length-1,n="https://reactjs.org/docs/error-decoder.html?invariant="+e,r=0;r<t;r++)n+="&args[]="+encodeURIComponent(arguments[r+1]);!function(e,t,n,r,o,i,u,a){if(!e){if(e=void 0,void 0===t)e=Error("Minified exception occurred; use the non-minified dev environment for the full error message and additional helpful warnings.");else{var c=[n,r,o,i,u,a],f=0;(e=Error(t.replace(/%s/g,function(){return c[f++]}))).name="Invariant Violation"}throw e.framesToPop=1,e}}(!1,"Minified React error #"+e+"; visit %s for the full message or use the non-minified dev environment for full errors and additional helpful warnings. ",n)}var g={isMounted:function(){return!1},enqueueForceUpdate:function(){},enqueueReplaceState:function(){},enqueueSetState:function(){}},w={};function j(e,t,n){this.props=e,this.context=t,this.refs=w,this.updater=n||g}function O(){}function S(e,t,n){this.props=e,this.context=t,this.refs=w,this.updater=n||g}j.prototype.isReactComponent={},j.prototype.setState=function(e,t){"object"!=typeof e&&"function"!=typeof e&&null!=e&&m("85"),this.updater.enqueueSetState(this,e,t,"setState")},j.prototype.forceUpdate=function(e){this.updater.enqueueForceUpdate(this,e,"forceUpdate")},O.prototype=j.prototype;var x=S.prototype=new O;x.constructor=S,r(x,j.prototype),x.isPureReactComponent=!0;var _={current:null},C={current:null},E=Object.prototype.hasOwnProperty,k={key:!0,ref:!0,__self:!0,__source:!0};function R(e,t,n){var r=void 0,o={},u=null,a=null;if(null!=t)for(r in void 0!==t.ref&&(a=t.ref),void 0!==t.key&&(u=""+t.key),t)E.call(t,r)&&!k.hasOwnProperty(r)&&(o[r]=t[r]);var c=arguments.length-2;if(1===c)o.children=n;else if(1<c){for(var f=Array(c),l=0;l<c;l++)f[l]=arguments[l+2];o.children=f}if(e&&e.defaultProps)for(r in c=e.defaultProps)void 0===o[r]&&(o[r]=c[r]);return{$$typeof:i,type:e,key:u,ref:a,props:o,_owner:C.current}}function P(e){return"object"==typeof e&&null!==e&&e.$$typeof===i}var U=/\/+/g,$=[];function L(e,t,n,r){if($.length){var o=$.pop();return o.result=e,o.keyPrefix=t,o.func=n,o.context=r,o.count=0,o}return{result:e,keyPrefix:t,func:n,context:r,count:0}}function A(e){e.result=null,e.keyPrefix=null,e.func=null,e.context=null,e.count=0,10>$.length&&$.push(e)}function M(e,t,n){return null==e?0:function e(t,n,r,o){var a=typeof t;"undefined"!==a&&"boolean"!==a||(t=null);var c=!1;if(null===t)c=!0;else switch(a){case"string":case"number":c=!0;break;case"object":switch(t.$$typeof){case i:case u:c=!0}}if(c)return r(o,t,""===n?"."+T(t,0):n),1;if(c=0,n=""===n?".":n+":",Array.isArray(t))for(var f=0;f<t.length;f++){var l=n+T(a=t[f],f);c+=e(a,l,r,o)}else if(l=null===t||"object"!=typeof t?null:"function"==typeof(l=b&&t[b]||t["@@iterator"])?l:null,"function"==typeof l)for(t=l.call(t),f=0;!(a=t.next()).done;)c+=e(a=a.value,l=n+T(a,f++),r,o);else"object"===a&&m("31","[object Object]"==(r=""+t)?"object with keys {"+Object.keys(t).join(", ")+"}":r,"");return c}(e,"",t,n)}function T(e,t){return"object"==typeof e&&null!==e&&null!=e.key?function(e){var t={"=":"=0",":":"=2"};return"$"+(""+e).replace(/[=:]/g,function(e){return t[e]})}(e.key):t.toString(36)}function I(e,t){e.func.call(e.context,t,e.count++)}function N(e,t,n){var r=e.result,o=e.keyPrefix;e=e.func.call(e.context,t,e.count++),Array.isArray(e)?D(e,r,n,function(e){return e}):null!=e&&(P(e)&&(e=function(e,t){return{$$typeof:i,type:e.type,key:t,ref:e.ref,props:e.props,_owner:e._owner}}(e,o+(!e.key||t&&t.key===e.key?"":(""+e.key).replace(U,"$&/")+"/")+n)),r.push(e))}function D(e,t,n,r,o){var i="";null!=n&&(i=(""+n).replace(U,"$&/")+"/"),M(e,N,t=L(t,i,r,o)),A(t)}function B(){var e=_.current;return null===e&&m("321"),e}var q={Children:{map:function(e,t,n){if(null==e)return e;var r=[];return D(e,r,null,t,n),r},forEach:function(e,t,n){if(null==e)return e;M(e,I,t=L(null,null,t,n)),A(t)},count:function(e){return M(e,function(){return null},null)},toArray:function(e){var t=[];return D(e,t,null,function(e){return e}),t},only:function(e){return P(e)||m("143"),e}},createRef:function(){return{current:null}},Component:j,PureComponent:S,createContext:function(e,t){return void 0===t&&(t=null),(e={$$typeof:s,_calculateChangedBits:t,_currentValue:e,_currentValue2:e,_threadCount:0,Provider:null,Consumer:null}).Provider={$$typeof:l,_context:e},e.Consumer=e},forwardRef:function(e){return{$$typeof:d,render:e}},lazy:function(e){return{$$typeof:v,_ctor:e,_status:-1,_result:null}},memo:function(e,t){return{$$typeof:h,type:e,compare:void 0===t?null:t}},useCallback:function(e,t){return B().useCallback(e,t)},useContext:function(e,t){return B().useContext(e,t)},useEffect:function(e,t){return B().useEffect(e,t)},useImperativeHandle:function(e,t,n){return B().useImperativeHandle(e,t,n)},useDebugValue:function(){},useLayoutEffect:function(e,t){return B().useLayoutEffect(e,t)},useMemo:function(e,t){return B().useMemo(e,t)},useReducer:function(e,t,n){return B().useReducer(e,t,n)},useRef:function(e){return B().useRef(e)},useState:function(e){return B().useState(e)},Fragment:a,StrictMode:c,Suspense:y,createElement:R,cloneElement:function(e,t,n){null==e&&m("267",e);var o=void 0,u=r({},e.props),a=e.key,c=e.ref,f=e._owner;if(null!=t){void 0!==t.ref&&(c=t.ref,f=C.current),void 0!==t.key&&(a=""+t.key);var l=void 0;for(o in e.type&&e.type.defaultProps&&(l=e.type.defaultProps),t)E.call(t,o)&&!k.hasOwnProperty(o)&&(u[o]=void 0===t[o]&&void 0!==l?l[o]:t[o])}if(1===(o=arguments.length-2))u.children=n;else if(1<o){l=Array(o);for(var s=0;s<o;s++)l[s]=arguments[s+2];u.children=l}return{$$typeof:i,type:e.type,key:a,ref:c,props:u,_owner:f}},createFactory:function(e){var t=R.bind(null,e);return t.type=e,t},isValidElement:P,version:"16.8.6",unstable_ConcurrentMode:p,unstable_Profiler:f,__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED:{ReactCurrentDispatcher:_,ReactCurrentOwner:C,assign:r}},V={default:q},z=V&&q||V;e.exports=z.default||z},22:function(e,t){e.exports=function(e){var t="undefined"!=typeof window&&window.location;if(!t)throw new Error("fixUrls requires window.location");if(!e||"string"!=typeof e)return e;var n=t.protocol+"//"+t.host,r=n+t.pathname.replace(/\/[^\/]*$/,"/");return e.replace(/url\s*\(((?:[^)(]|\((?:[^)(]+|\([^)(]*\))*\))*)\)/gi,function(e,t){var o,i=t.trim().replace(/^"(.*)"$/,function(e,t){return t}).replace(/^'(.*)'$/,function(e,t){return t});return/^(#|data:|http:\/\/|https:\/\/|file:\/\/\/|\s*$)/i.test(i)?e:(o=0===i.indexOf("//")?i:0===i.indexOf("/")?n+i:r+i.replace(/^\.\//,""),"url("+JSON.stringify(o)+")")})}},252:function(e,t,n){var r=n(253);"string"==typeof r&&(r=[[e.i,r,""]]);var o={hmr:!0,transform:void 0,insertInto:void 0};n(11)(r,o);r.locals&&(e.exports=r.locals)},253:function(e,t,n){(e.exports=n(10)(!1)).push([e.i,".Drawer .show-drawer {\n    /* width: 400px; */\n    height: 100%;\n    background: #fff;\n    position: absolute;\n    top: 0;\n    z-index: 500;\n    transition: all .7s cubic-bezier(0.35, 0, 0.25, 1);\n}\n\n.Drawer .operate-drawer {\n    width: 60px;\n    height: 100%;\n    position: absolute;\n    top: 0;\n    z-index: 500;\n    transition: all .7s cubic-bezier(0.35, 0, 0.25, 1);\n    /* background: linear-gradient(to right, rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0)); */\n    display: flex;\n    align-items: center;\n    justify-content: center;\n}\n\n.Drawer .operate-drawer .scale-wrapper{\n    transition: all .7s cubic-bezier(0.35, 0, 0.25, 1);\n}",""])}})});