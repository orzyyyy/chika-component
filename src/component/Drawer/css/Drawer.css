.Drawer .show-drawer {
    width: 400px;
    height: 100%;
    background: #f96;
    position: absolute;
    top: 0;
    z-index: 500;
    transition: all .7s cubic-bezier(0.35, 0, 0.25, 1);
}

.Drawer .operate-drawer {
    width: 60px;
    height: 100%;
    position: absolute;
    top: 0;
    z-index: 500;
    transition: all .7s cubic-bezier(0.35, 0, 0.25, 1);
    /* background: linear-gradient(to right, rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0)); */
    display: flex;
    align-items: center;
    justify-content: center;
}

.Drawer .operate-drawer .scale-wrapper{
    transition: all .7s cubic-bezier(0.35, 0, 0.25, 1);
}