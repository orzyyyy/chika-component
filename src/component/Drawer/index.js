/*
 * @Author: zy9@github.com/zy410419243
 * @Date: 2018-09-14 13:54:25
 * @Last Modified by: zy9
 * @Last Modified time: 2018-09-22 09:18:24
 */
import React, { Component } from 'react';

import './css/Drawer.css';

export default class Drawer extends Component {
	handlePosition = (direction, clientView, visible, width, type) => {
		const transform = this.handleTransform(direction);
		const { view } = this.handleClientView(direction, clientView, visible, width, type);

		return `${ transform }(${ view }px)`;
	}

	handleTransform = direction => {
		if(direction == 'left' || direction == 'right') {
			return 'translateX';
		} else if(direction == 'top' || direction == 'bottom') {
			return 'translateY';
		}

		return 'transform error';
	}

	// 返回箭头方向值跟抽屉位置
	handleClientView = (direction, clientView, visible, width, type) => {
		let view, iconDeg;

		if(direction == 'left' || direction == 'right') {
			clientView = clientView || (document.documentElement.clientWidth || document.body.clientWidth);
		} else if(direction == 'top' || direction == 'bottom') {
			clientView = clientView || (document.documentElement.clientHeight || document.body.clientHeight);
		}

		if(direction == 'left' || direction == 'top') {
			if(type == 'operate') {
				view = visible ? width : 0;
			} else {
				view = visible ? 0 : -width;
			}

			iconDeg = visible ? 0 : 180;
		} else if(direction == 'right' || direction == 'bottom') {
			if(type == 'operate') {
				view = visible ? clientView - width - 60 : clientView - 60;
			} else {
				view = visible ? clientView - width : clientView;
			}

			iconDeg = visible ? 180 : 0;
		}

		return { view, iconDeg };
	}

	handleOperateClick = () => {
		const { onChange, visible } = this.props;
		let iconDeg = visible ? 180 : 0;

		onChange && onChange(!visible);

		this.setState({ iconDeg });
	}

    render = () => {
    	const { children, width = (document.body.clientWidth || document.documentElement.clientWidth) * 0.7, operaNode, direction = 'left', visible, isOperateShow = true, clientView, style } = this.props;
    	const { iconDeg } = this.handleClientView(direction, clientView, visible, width);

    	const operateDrawer = (
    		<div className='operate-drawer' style={{ transform: this.handlePosition(direction, clientView, visible, width, 'operate'), background: `linear-gradient(to ${ direction == 'left' ? 'right' : 'left' }, rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0))` }} onClick={ this.handleOperateClick }>
    			<div className='scale-wrapper' style={{ transform: `rotateZ(${ iconDeg }deg)` }}>
    				{ operaNode ? operaNode : <img src='../../assets/Drawer/arrow.png' /> }
    			</div>
    		</div>
    	);

    	const showDrawer = (
    		<div className='show-drawer' style={ Object.assign({}, { width }, { transform: this.handlePosition(direction, clientView, visible, width) }, style) }>
    			{ children }
    		</div>
    	);

    	return (
    		<div className='Drawer'>
    			{ showDrawer }

    			{ isOperateShow && operateDrawer }
    		</div>
    	);
    }
}