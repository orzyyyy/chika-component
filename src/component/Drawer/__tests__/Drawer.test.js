/*
 * @Author: zy9@github.com/zy410419243
 * @Date: 2018-09-21 09:03:47
 * @Last Modified by: zy9
 * @Last Modified time: 2018-09-22 09:34:27
 */
import React from 'react';
import { render, mount, shallow } from 'enzyme';
import Drawer from '..';

describe('Drawer', () => {
	it('render correctly', () => {
		const wrapperShow = render(
			<Drawer visible={ true }>test</Drawer>
		);

		const wrapperHide = render(
			<Drawer visible={ false }>test</Drawer>
		);

		expect(wrapperShow).toMatchSnapshot();
		expect(wrapperHide).toMatchSnapshot();
	});

	it('when operate drawer is clicked, onChange should to be called', () => {
		const onChange = jest.fn();
		const wrapper = mount(
			<Drawer visible={ true } onChange={ onChange } direction='right'>test</Drawer>
		);

		wrapper.find('.operate-drawer').simulate('click');

		expect(onChange).toBeCalled();
	});
});