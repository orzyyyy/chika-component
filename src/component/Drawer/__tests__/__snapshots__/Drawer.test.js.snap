// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer render correctly 1`] = `
<div
  class="Drawer"
>
  <div
    class="show-drawer"
    style="width:0;transform:translateX(0px)"
  >
    test
  </div>
  <div
    class="operate-drawer"
    style="transform:translateX(0px);background:linear-gradient(to right, rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0))"
  >
    <div
      class="scale-wrapper"
      style="transform:rotateZ(0deg)"
    >
      <img
        src="../../assets/Drawer/arrow.png"
      />
    </div>
  </div>
</div>
`;

exports[`Drawer render correctly 2`] = `
<div
  class="Drawer"
>
  <div
    class="show-drawer"
    style="width:0;transform:translateX(0px)"
  >
    test
  </div>
  <div
    class="operate-drawer"
    style="transform:translateX(0px);background:linear-gradient(to right, rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0))"
  >
    <div
      class="scale-wrapper"
      style="transform:rotateZ(180deg)"
    >
      <img
        src="../../assets/Drawer/arrow.png"
      />
    </div>
  </div>
</div>
`;
